src/__tests__/product.spec.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable consistent-return */
+/* eslint-disable node/no-unpublished-require */
+const supertest = require('supertest');
+const app = require('../app');
+const UserModel = require('../models/user');
+const ProductModel = require('../models/product');
+const connectDatabase = require('../db/connection');
+
+const createCurrentUser = {
+  firstName: 'adnan',
+  lastName: 'khaldar',
+  email: 'adnan1@outlook.com',
+  username: 'adnan1',
+  donated: [],
+  requested: [],
+  provider: 'Local',
+  providerId: 'Local',
+  passwordHash: '$2b$10$R5NUgaHK51jYdi59ncmwue/lorlCHturbAmFxJ02cS38eumzNSx7O',
+};
+
+const createRequestPost = {
+  title: 'test',
+  description: 'test',
+  photos: ['test'],
+  category: 'Other',
+  location: 'test',
+  productCondition: 'New',
+  shippingOptions: 'To be determined',
+  postType: 'Request',
+};
+
+const createDonatePost = {
+  title: 'test',
+  description: 'test',
+  photos: ['test'],
+  category: 'Other',
+  location: 'test',
+  productCondition: 'New',
+  shippingOptions: 'To be determined',
+  postType: 'Donate',
+};
+
+let newUser;
+let token;
+
+let requestPostId;
+let donatePostId;
+
+beforeAll(async () => {
+  connectDatabase();
+  await ProductModel.deleteMany();
+
+  newUser = await UserModel.create(createCurrentUser);
+});
+
+describe('POST /api/product && GET /api/product && DELETE /api/product/:id', () => {
```

Comment:
this describe is about all the it() properties below so it should not contain get and delete parts i think.